pkg/util/util.go

```diff
@@ -61,6 +63,53 @@ func loadCertsFromPaths(paths []string, pool *x509.CertPool) (*x509.CertPool, er
 	return pool, nil
 }
 
+func getClientCertificates(certFile, keyFile string) ([]tls.Certificate, error) {
+	if _, err := os.Stat(certFile); os.IsNotExist(err) {
+		return nil, fmt.Errorf("unable to locate certificate %s: %w", certFile, err)
+	}
+
+	if _, err := os.Stat(keyFile); os.IsNotExist(err) {
+		return nil, fmt.Errorf("unable to locate key %s: %w", keyFile, err)
+	}
+
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse cert/key pair %s, %s: %w", certFile, keyFile, err)
+	}
+
+	return []tls.Certificate{cert}, nil
+}
+
+func GetHTTPClient(certFile, keyFile string, insecureSkipVerify bool, useSystemPool bool, caFiles ...string) (*http.Client, error) {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
```

Comment:
Why is the clone necessary here?

Reply:
Just trying to avoid mutating the default transport and instead work on a copy of it. Happy to remove the clone though.